src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -5260,13 +5259,11 @@ QList<QgsRelation> QgsPostgresProvider::discoverRelations( const QgsVectorLayer
       refSchema = refSchema.mid( 1, refSchema.length() - 2 );
     }
     const QString refColumn = sqlResult.PQgetvalue( row, 4 );
-    const QString position = sqlResult.PQgetvalue( row, 5 );
-    const QList<QgsVectorLayer *> foundLayers = searchLayers( layers, mUri.connectionInfo( false ), refSchema, refTable );
-    if ( ( position == QLatin1String( "1" ) ) || ( !refTableFound.contains( refTable ) ) )
+    // try to find if we have layers for the referenced table
+    const auto foundLayers = searchLayers( layers, mUri.connectionInfo( false ), refSchema, refTable );
```

Comment:
this makes things less readable, let's not change it:

```suggestion
    const QList<QgsVectorLayer *> foundLayers = searchLayers( layers, mUri.connectionInfo( false ), refSchema, refTable );
```